staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -375,19 +377,27 @@ func (d *DiscoveryClient) SwaggerSchema(version schema.GroupVersion) (*swagger.A
 	return &schema, nil
 }
 
-// OpenAPISchema fetches the open api schema using a rest client and parses the json.
-// Warning: this is very expensive (~1.2s)
-func (d *DiscoveryClient) OpenAPISchema() (*spec.Swagger, error) {
-	data, err := d.restClient.Get().AbsPath("/swagger.json").Do().Raw()
+// OpenAPISchema fetches the open api schema using a rest client and parses the proto.
+func (d *DiscoveryClient) OpenAPISchema() (*openapi_v2.Document, error) {
+	stream, err := d.restClient.Get().AbsPath("/swagger-2.0.0.pb-v1.gz").Stream()
```

Comment:
This looks off to me.  Why wouldn't you request a gzipped encoding of the normal protobuf struct?